services/code_hosting/git_hub_service.rb

```diff
@@ -1,131 +1,409 @@
+      end
+
+      def temporary_git_storage(root_path = Dir.tmpdir)
+        temp_storage = File.join(root_path, ".tmp")
+        @temporary_git_storage ||= temp_storage
+        FileUtils.mkdir_p(@temporary_git_storage) unless File.directory?(@temporary_git_storage)
+        return @temporary_git_storage
+      end
+
+      # This method shallow clones the repo using the provider credential and looks for a valid _fastlane_
+      # configuration, parses the Fastfile (if found), and returns the lanes in the following way:
+      #  {
+      #    "no_platform": {
+      #      "": {
+      #        "description": [
+      #
+      #        ],
+      #        "actions": [
+      #          {
+      #            "action": "default_platform",
+      #            "parameters": "ios"
+      #          }
+      #        ]
+      #      }
+      #    },
+      #    "ios": {
+      #      "test": {
+      #        "description": [
+      #          "Description of what the lane does"
+      #        ],
+      #        "actions": [
+      #          {
+      #            "action": "gym",
+      #            "parameters": {
+      #              "skip_package_ipa": false,
+      #              "clean": true,
+      #              "project": "../fastlane-ci-demoapp.xcodeproj",
+      #              "scheme": "fastlane-ci-demoapp"
+      #            }
+      #          }
+      #        ],
+      #        "private": false
+      #      }
+      #    }
+      #  }
+      #
+      # @param repo_url [String]
+      # @param branch [String]
+      # @param path [String]
+      # @param provider_credential [GithubProviderCredential]
+      # @return [Hash<String, Hash>] being the key the name of the platform (:ios, :android, :no_platform) and the Hash the underlying lane (name, actions).
+      def peek_fastfile_configuration(repo_url: nil, branch: nil, provider_credential: nil, path: self.temp_path, cache: true)
+        repo = repo_from_url(repo_url)
+        return self.cache[[repo, branch].join("/")] if cache && self.cache && !self.cache[[repo, branch].join("/")].nil? && self.cache[[repo, branch].join("/")].kind_of?(Hash)
+        path = File.join(path, repo, branch)
+        begin
+          git_path = File.join(path, repo.split("/").last)
+          # This triggers the check of an existing repo in the given path,
+          # we recover from the error making the clone and checkout
+          Git.open(git_path)
+          fastfile_path = self.fastfile_path(root_path: git_path)
+          fastfile = Fastlane::FastfileParser.new(path: fastfile_path)
+          fastfile_config = {}
+          # TODO: FastfileParser provides a nice and clean tree of the Fastfile given,
+          # but there's a specific case when there might be lanes or actions out of
+          # a platform scope, in those cases the key is nil. We sanitize that so the user
+          # still can select lanes without platform. But this should be done from the
+          # FastfileParser side.
```

Comment:
Suggest linking an issue here for the `FastfileParser` repo

Reply:
Will open an issue on this, ideally, `FastfileParser` will generate a `to_json` nicely, so the processing below isn't needed

Reply:
https://github.com/fastlane/fastfile-parser/issues/8